### PR TITLE
Neuronexus datetime parsing

### PR DESCRIPTION
**Pull request by h-mayorquin**
Follow up on this:

#3373

From this [comment](https://github.com/NeuralEnsemble/python-neo/pull/1509#discussion_r1777624427) I did not have enough time to address there

Just a more standard way of parsing datetime strings @zm711

### CODE REVIEW

**Comment by zm711** [comment]
Thanks Sam told me something like that existed, but he couldn't remember so he said we could merge it and that you would swoop in and save the day :)

**Comment by zm711** [comment]
Interesting it says it's not isoformat?

**Comment by h-mayorquin** [comment]
No, that's iso format. It seems that support for reading iso datetimes was improved on python 3.11. This works just fine locally for me.

More worrisome I discovered that your test on python 3.11 are not really on python 3.11:

image

**Comment by zm711** [comment]
I think we must've had a bad cache and it kept using the bad cache because it only updates if the cache is wrong in the action. I guess let me remove the cache and we can try again without the cache.

File: neo/rawio/neuronexusrawio.py [hunk 1]
```diff
@@ -219,21 +219,10 @@ def _parse_header(self):
...
-        hour = int(time_info[0][-2:])
-        minute = int(time_info[1])
-        second = int(float(time_info[2]))
-        microsecond = int(1000 * 1000 * (float(time_info[2]) - second))# second -> micro is 1000 * 1000
-
-        rec_datetime = datetime.datetime(year, month, day, hour, minute, second, microsecond)
+        # date comes out as: '2024-07-01T13:04:49.4972245-04:00' so in ISO format
+        datetime_string = self.metadata["status"]["start_time"]
+        rec_datetime = datetime.datetime.fromisoformat(datetime_string)
+
```
Comment (zm711) [inline neo/rawio/neuronexusrawio.py:225]: what if we do
try-except where we cut off the trailing -4:00 which I think was support < 3.11.

**Comment by zm711** [comment]
Cache is now fixed!

File: neo/rawio/neuronexusrawio.py [hunk 2]
```diff
@@ -219,21 +221,17 @@ def _parse_header(self):
...
-        second = int(float(time_info[2]))
-        microsecond = int(1000 * 1000 * (float(time_info[2]) - second))# second -> micro is 1000 * 1000
-
-        rec_datetime = datetime.datetime(year, month, day, hour, minute, second, microsecond)
+        # date comes out as: '2024-07-01T13:04:49.4972245-04:00' so in ISO format
+        datetime_string = self.metadata["status"]["start_time"]
+        
+
+        # TODO: remove once minimal version is 3.11
+        # Python 3.10 and below do not support the fractional seconds in fromisoformat
```
Comment (zm711) [inline neo/rawio/neuronexusrawio.py:229]: great comments thanks!

**Review by zm711 — APPROVED** [review]

**Comment by zm711** [comment]
Thanks Heberto!

**Comment by zm711** [comment]
And thanks for the learning experience!

**Pull request merged**